micro_http/src/connection.rs

```diff
@@ -0,0 +1,634 @@
+                            };
+
+                            // Update the `line_start_index` to where we finished parsing.
+                            line_start_index = line_end_index + CRLF_LEN;
+                        }
+                        // If we have an incomplete header line.
+                        None => {
+                            // If we have parsed 1024 bytes and still haven't found the header
+                            // line end sequence.
+                            if line_start_index == 0 && bytes_read == 1024 {
+                                // Header line is longer than 1024 bytes, so it is invalid.
+                                return Err(ConnectionError::ParsingError(
+                                    RequestError::InvalidHeader,
+                                ));
+                            }
+                            // Move the incomplete header line from the end of the buffer to
+                            // the beginning, so that we can append the rest of the line and
+                            // parse it in the next `try_read` call.
+                            self.reset_buffer(line_start_index, bytes_read);
+                            break;
+                        }
+                    }
+                }
+                ConnectionState::WaitingForBody => {
+                    // If what we have just read is not enough to complete the request and
+                    // there are more bytes pertaining to the body of the request.
+                    if self.body_bytes_to_be_read > bytes_read as i32 - line_start_index as i32 {
+                        // Append everything that we read to our current incomplete body and update
+                        // `body_bytes_to_be_read`.
+                        self.body_vec
+                            .extend_from_slice(&self.buffer[line_start_index..bytes_read]);
+                        self.body_bytes_to_be_read -= bytes_read as i32 - line_start_index as i32;
+
+                        // Clear the buffer.
+                        for i in 0..1024 {
+                            self.buffer[i] = 0;
+                        }
+
+                        break;
+                    } else {
+                        // Append only the remaining necessary bytes to the body of the request.
+                        self.body_vec.extend_from_slice(
+                            &self.buffer[line_start_index
+                                ..(line_start_index + self.body_bytes_to_be_read as usize)],
+                        );
+                        line_start_index += self.body_bytes_to_be_read as usize;
+                        self.body_bytes_to_be_read = 0;
+                    }
+
+                    let request = self.request.as_mut().unwrap();
+                    // If there are no more bytes to be read for this request.
+                    if self.body_bytes_to_be_read == 0 {
```

Comment:
This `if` is always `true`. The previous block either `break`s or sets `self.body_bytes_to_be_read = 0`